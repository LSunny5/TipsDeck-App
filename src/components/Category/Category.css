/*container for category buttons*/
.categoryBox {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 5% auto 10%;
}

/*css for each category button design*/
.categoryButton {
    font-size: 2em;
    border: 1px solid black;
    color: black;
    margin: 3% 4%;
    padding: 6%;
}

/*css for Link or name of category buttons*/
.categoryName {
    text-decoration: none;
    width: 70vw;
}

.categoryName:hover {
    background-color: teal;
}

.categoryName:active {
    background-color: lightblue;
}

/* for tablets */
@media only screen and (min-width: 600px) { 
    .Categories > h1 {
        font-size: 2em;
    }

    .Categories > h2 {
        font-size: 1em;
    }
    
    .categoryBox {
       flex-direction: row;
       width: 95%;
       flex-wrap: wrap;
       margin: 3% auto;
       justify-content: space-around;
       font-size: 0.6em;
   }
   
   .categoryButton {
       margin: 3%;
       padding: 20%;
   }
   
   .categoryName {
       width: 30vw;
   }
}

/* for large screens */
@media only screen and (min-width: 1024px) {
    .categoryBox {
        width: 90%;
        margin: 3% auto;
    }
    
    .categoryButton {
        margin: 1rem;
        padding: 3rem;
    }
    
    .categoryName {
        width: 25%;
    }
}