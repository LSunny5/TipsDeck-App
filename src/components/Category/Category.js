import React from 'react';
import './Category.css';
import { Link } from 'react-router-dom';
import TipsDeckContext from '../../TipsDeckContext';

//Shows a list of categories a user may select from
class Category extends React.Component {
    static contextType = TipsDeckContext;

    render() {
        const { categories = [] } = this.context;
        return (
            <section className="Categories">
                <h1>Categories</h1>
                <h2>Choose a category </h2>
                <div className="categoryBox">
                    {/* map out all the categories from the file/database */}
                    {(categories.length > 0) ? (
                        categories.map(category =>
                            <Link key={category.id}
                                to={{
                                    pathname: `/Category/${category.name}`, 
                                    state: category.name
                                }}
                                className="categoryName">
                                <div className="categoryButton">
                                    {category.name}
                                </div>
                            </Link>
                        )) : (
                            <div className="noCategories">
                                No categories found
                            </div>)}
                </div>
            </section>
        );
    };
}

export default Category;