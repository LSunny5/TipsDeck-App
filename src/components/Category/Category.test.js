import React from 'react';
import ReactDOM from 'react-dom';
import Category from './Category';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`Category component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Category />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the Category component', () => {
        const wrapper = shallow(<Category />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});