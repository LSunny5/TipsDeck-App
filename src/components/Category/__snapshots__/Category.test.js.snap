// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component renders the Category component 1`] = `
<section
  className="Categories"
>
  <h1>
    Categories
  </h1>
  <h2>
    Choose a category 
  </h2>
  <div
    className="categoryBox"
  >
    <div
      className="noCategories"
    >
      No categories found
    </div>
    <Link
      className="categoryName"
      to="/Random"
    >
      <div
        className="categoryButton"
      >
        Random
      </div>
    </Link>
  </div>
</section>
`;
